### ISSUE TITLE
[Enhancement]

### ISSUE THREAD

**Issue by perdolka** [Lej77/tab-unloader-for-tree-style-tab#2]
1 add timer for automatic suspend
2 add whitelist

**Comment by Lej77** [comment]
@perdolka I might implement something like that but is it really needed when there are other [extensions that also support Tree Style Tab](https://github.com/piroor/treestyletab#addons-extend-tst) which implement this such as [Auto Tab Discard by richard_neomy](https://addons.mozilla.org/firefox/addon/auto-tab-discard/).

**Comment by perdolka** [comment]
Auto Tab Discard still has some old issues with TST:
https://github.com/rNeomy/auto-tab-discard/issues/26
https://github.com/rNeomy/auto-tab-discard/issues/28

UnloadTabs - just doesn't work for me for some reason.

Anyway, it would be great to to add at least "suspend the whole tree".
Thanks for the great addon.

**Comment by Lej77** [comment]
@perdolka 
https://github.com/rNeomy/auto-tab-discard/issues/26 seems to be the same issue as what you reported in https://github.com/Lej77/tab-unloader-for-tree-style-tab/issues/3. So this isn't something that should make this extension better than Auto Tab Discard.

https://github.com/rNeomy/auto-tab-discard/issues/28 should be fixed with https://github.com/rNeomy/auto-tab-discard/pull/49.

[UnloadTabs](https://addons.mozilla.org/firefox/addon/unload-tabs/) should work. Did you go to its options page and check the `Dim unloaded Tabs and add context menu options.` checkbox?

I'll keep your requests in mind when I work more on this extension.

**Comment by perdolka** [comment]
> Did you go to its options page and check the Dim unloaded Tabs and add context menu options. checkbox?

I didn't notice this option. Thanks.

**Comment by Lej77** [comment]
@perdolka I have released a new version (v5.0) and it includes a context menu item to unload a whole tree. It is disabled by default though so you have to open the options page and activate it. The option is located in the `Context menu items` section in the sub section `Unload tree`. In that sub-section select the `Add a context menu item in Tree Style Tab's context menu to unload a tab tree.` checkbox and the item will be added to the context menu.

**Comment by perdolka** [comment]
Thank you!

**Comment by dxlr8r** [comment]
> add timer for automatic suspend

That would be awesome. Then I could finally switch to Quantum!

**Comment by Lej77** [comment]
@dxlr8r Any reason why you can't use [Auto Tab Discard by richard_neomy](https://addons.mozilla.org/firefox/addon/auto-tab-discard/) for automatic suspend?

**Comment by Lej77** [comment]
@dxlr8r We use the same API's yes but most issues are probably because of the implementation, not the API's, so if/when I implement auto discarding it should not have the same problem.

**Comment by dsl101** [comment]
I'd love an option in the main Left / Middle / Right settings to unload a whole tree if the tab clicked has children. The context menu is OK, but much more cumbersome to use, and I'd like to have unload tree as a default. I can't currently see a way to do that?

**Comment by Lej77** [comment]
@dsl101 You are right that there is no way to currently unload a tree when Left / Middle / Right clicking a tab but that is a good idea. If I get some time over I'll see about implementing it!

**Comment by Lej77** [comment]
@dsl101 I just released version 6.0 of the extension and it has some support for unloading trees when clicking on a tab. It now has a new checkbox for each mouse button's section that says `Unload Tree: if the tab that should be unloaded has child tabs then those will be unloaded as well.` and will switch that mouse button from unloading a single tab to unloading a whole tree.

**Comment by dsl101** [comment]
Fab—will check it out now :). Thanks!

**Comment by dsl101** [comment]
Great—that works well, apart from in one common (for me) scenario. I create 'folders' using the TST built in URL like this:
```
moz-extension://9b563068-d3b7-4d91-8d34-26d99b7af13b/resources/group-tab.html?title=Work
```
This is a really useful way of keeping tabs organised, but I rarely, if ever, click on the actual folder itself (just use the open/close icon to expand & collapse it). I have Unload Tab assigned to Alt+Left Click, but if I do that on an already unloaded folder, nothing happens to the tree of tabs underneath.

Would it be possible, in this case, even though the tab being clicked on is already unloaded and therefore doesn't need its state changing, your extension could still check for potentially loaded child tabs and unload them?

Thanks!

**Comment by Lej77** [comment]
@dsl101 Yeah, that sounds like a bug. I'll try to fix that soon!

**Comment by Lej77** [comment]
@dsl101 I release version 6.1 of the extension which includes a fix for the issue you noticed, so it should work correctly now.

**Comment by i-need-to-tell-you-something** [comment]
I think automatic tab suspension is now built into Firefox and this issue can be closed, no? And I feel automatic tab unloading should be a separate addon anyway - in synergy with - rather than a part of this addon.

**Comment by dsl101** [comment]
@i-need-to-tell-you-something I think that is 2 different things. I wasn't looking for automatic tab suspension, but the option to explicitly suspend a 'tree' of tabs. The current solution from @Lej77 is working well for me though, so I suspect you're right that this can be closed.
